### PR TITLE
add instance_info module

### LINKED ISSUE #49: [Request] Create instance_info module
Create an instance_info module to return basic information about a SQL instance on a host.

### PR DESCRIPTION
**Pull request by lowlydba**
## Description

Add module to give high-level instance details.

## How Has This Been Tested?

int tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) - Fixes #
- [x] New feature (non-breaking change which adds functionality) - Resolves #49 

## Checklist:

- [x] I have read/followed the [**CONTRIBUTING**](https://github.com/LowlyDBA/lowlydba.sqlserver/blob/main/.github/CONTRIBUTING.md) document.
- [x] I have read/followed the [PR Quick Start Guide](https://github.com/ansible/community-docs/blob/main/create_pr_quick_start_guide.rst)  # TODO: update with link to published doc
- [x] I have added tests to cover my changes or they are N/A.
- [x] New module options/parameters include a [`version_added` property](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#documentation-fields).

### CODE REVIEW

File: changelogs/fragments/51-add-instance_info-module.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - Added instance_info module
```
Comment (briantist) [inline changelogs/fragments/51-add-instance_info-module.yml:2]: New modules shouldn't have a (manual) changelog entry, they'll automatically have changelog entries generated by `antsibull-changelog`, so a fragment is only needed if there are other changes in the PR that should be reported.

File: galaxy.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 namespace: lowlydba
 name: sqlserver
-version: 0.1.1
+version: 0.2.0
```
Comment (briantist) [inline galaxy.yml:5]: This is fine here though I'd usually make it in a separate commit (I would typically push this directly right after a release instead of making a PR).

File: plugins/modules/instance_info.ps1 [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    $getSplat = @{
+        SqlInstance = $sqlInstance
+        SqlCredential = $sqlCredential
+        EnableException = $true
+    }
+    $output = Get-DbaConnection @getSplat | Select-Object -Property "ComputerName", "SqlInstance", "InstanceName" -First 1
+
+    if ($null -ne $output) {
+        # Add additional fields
+        $extraProperties = @("BuildNumber", "Language", "VersionMajor", "VersionMinor", "VersionString", "Collation", "ProductLevel", "IsClustered", "LoginMode")
```
Comment (briantist) [inline plugins/modules/instance_info.ps1:34]: re: the sanity long line failure, one of the things I like anyway about PowerShell's multi-line array and hashtable syntax is that the commas become optional that way:
```powershell
$a = @(
    "item1"
    "item2"
)

$h = @{
    KeyA = "one"
    KeyB = "two"
}
```
(I knew about the hashtable one for a while, took me forever to realize it worked for arrays too 🤦‍♂️)

Comment (lowlydba) [reply plugins/modules/instance_info.ps1:34]: TIL that for arrays!

File: galaxy.yml (hunk 2 above)
Comment (lowlydba) [reply galaxy.yml:5]: Yeah I realized I wasn't sure where best to do this - I was worried about putting the `version_added` and then later using a different version than `2.0` next 🤔
